docs/network-binding-plugin.md

```diff
@@ -0,0 +1,342 @@
+
+> **Note**: `changeDomain` is the function that accepts a domain and VMI spec,
+> returning a new mutated domain spec.
+
+```go
+import "kubevirt.io/api/core/v1"
+import hooksv1alpha2 "kubevirt.io/kubevirt/pkg/hooks/v1alpha2"
+import domainschema "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+
+type V1alpha2Server struct{}
+
+func (s V1alpha2Server) OnDefineDomain(_ context.Context, params *hooksv1alpha2.OnDefineDomainParams) (*hooksv1alpha2.OnDefineDomainResult, error) {
+    vmi := &v1.VirtualMachineInstance{}
+    if err := json.Unmarshal(params.GetVmi(), vmi); err != nil {
+        return nil, fmt.Errorf("failed to unmarshal VMI: %v", err)
+    }
+
+    domainSpec := &domainschema.DomainSpec{
+        // Unmarshalling domain spec makes the XML namespace attribute empty.
+        // Some domain parameters requires namespace to be defined.
+        // e.g: https://libvirt.org/drvqemu.html#pass-through-of-arbitrary-qemu-commands
+        XmlNS: libvirtDomainQemuSchema,
+    }
+    if err := xml.Unmarshal(domainXML, domainSpec); err != nil {
+        return nil, fmt.Errorf("failed to unmarshal given domain spec: %v", err)
+    }
+
+    updatedDomainSpec, err := changeDomain(domainSpec, vmi)
+    if err != nil {
+        return nil, err
+    }
+
+    newDomainXML, err := xml.Marshal(updatedDomainSpec)
+    if err != nil {
+        return nil, fmt.Errorf("failed to marshal updated domain spec: %v", err)
+    }
+
+    return &hooksv1alpha2.OnDefineDomainResult{
+        DomainXML: newDomainXML,
+    }, nil
+}
+```
+
+It is recommended to use the latest available hook version.
+The unused methods must be defined to satisfy the interface, preferably with
+a default non-error response.
+
+#### Pod interface name
+
+The name of the network interface in the pod, to which the relevant network
+is to bind to, is expected to be known.
+
+- For the primary network, the default `eth0` naming is expected.
+- For the secondary networks, Kubevirt explicitly specifies the name through
+  Multus annotations of the pod. By that it assumes the CNI plugin used
+  confirms with the CNI specification and will name the interface accordingly.
+  The name used is generated by Kubevirt with the following format:
+  `pod<network name hash ID>`.
+  - Network hash ID: The hash ID is generated based on the network name specified
+    in the VM/VMI interface & network spec.
+    To programmatically generate it, one can use the following code snippet:
+
+```go
+import "kubevirt.io/kubevirt/pkg/network/namescheme"
+
+ifaceName := namescheme.GenerateHashedInterfaceName(network_name)
+```
+
+#### Domain configuration
+
+Changing the domain configuration usually involves the addition of a device
+or a change in an existing device.
+
+Either way, it is important to understand which are the networking
+devices and what are the expectations from Kubevirt:
+
+- Domain API: "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
```

Comment:
nit: I would add new line after each bullet in this section.

Reply:
Done